### PR TITLE
Cleanup file based system access control

### PR DESCRIPTION
**Pull request by ksobolew**
Some cleanups and minor fixes to tests in and around `FileBasedSystemAccessControl`.

Extracted from #9176

### CODE REVIEW

File: lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/FileBasedSystemAccessControl.java [hunk 1]
```diff
@@ -155,31 +155,26 @@ private FileBasedSystemAccessControl(
         ImmutableSet.Builder<AnyCatalogPermissionsRule> anyCatalogPermissionsRules = ImmutableSet.builder();
```
Comment (kokosing) [inline lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/FileBasedSystemAccessControl.java:155]: Can you please improve commit message. The current one is very generic. Maybe `Leverage Optional::stream in FileBasedSystemAccessControl`?

File: lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/security/TestFileBasedSystemAccessControl.java [hunk 2]
```diff
@@ -112,6 +113,7 @@
     private static final String SET_SYSTEM_SESSION_PROPERTY_ACCESS_DENIED_MESSAGE = "Access Denied: Cannot set system session property .*";
     private static final String SET_CATALOG_SESSION_PROPERTY_ACCESS_DENIED_MESSAGE = "Access Denied: Cannot set catalog session property .*";
 
+    @SuppressWarnings("deprecation")
```
Comment (kokosing) [inline lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/security/TestFileBasedSystemAccessControl.java:116]: What and why do we suppress? Maybe we can fix it by not using deprecated API?

**Review by kokosing — APPROVED** [review]
% comments

**Review by gaurav8297 — APPROVED** [review]

File: lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/security/TestFileBasedSystemAccessControl.java (hunk 2 above)
Comment (ksobolew) [reply lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/security/TestFileBasedSystemAccessControl.java:116]: There is a `checkCanSetUser` method in `SystemAccessControl`, which is deprecated. The replacement is supposed to be `checkCanImpersonateUser`but it's not a direct replacement and I'm not yet sure what's the difference in their use cases.

Comment (kokosing) [reply lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/security/TestFileBasedSystemAccessControl.java:116]: Yeah... this let's not suppress this. We need to remember about `checkCanSetUser` and eventually remove that method.

Let's remove this suppression.

Comment (kokosing) [reply lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/security/TestFileBasedSystemAccessControl.java:116]: The proper fix is here: https://github.com/trinodb/trino/pull/4081

Comment (ksobolew) [reply lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/security/TestFileBasedSystemAccessControl.java:116]: OK, cool, I'll leave it then. Thanks!

**Comment by ksobolew** [comment]
Ouch... In `:trino-hive,:trino-orc`

```
Error:  java.lang.OutOfMemoryError: Java heap space
```

File: lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/security/TestFileBasedSystemAccessControl.java [hunk 3]
```diff
@@ -1188,6 +1190,7 @@ public void testRefreshing()
         accessControl.checkCanCreateView(alice, aliceView);
         accessControl.checkCanCreateView(alice, aliceView);
 
+        //noinspection UnstableApiUsage
```
Comment (kokosing) [inline lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/security/TestFileBasedSystemAccessControl.java:1193]: We should suggest to disable this inspection. A lot of guava is using `@Beta` that causing this inspection to trigger. Let's update https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md instead.

Comment (ksobolew) [reply lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/security/TestFileBasedSystemAccessControl.java:1193]: #9395

**Pull request merged**